### PR TITLE
netlink: use v2 cache api

### PR DESCRIPTION
**Pull request by toanju**
* update libnl3 dependency to 3.2.29
* drop all local caches of netlink events (just use the cache provided by libnl)
* minor cleanup

### CODE REVIEW

File: src/roflibs/netlink/cnetlink.cpp [hunk 1]
```diff
@@ -272,54 +260,21 @@ void cnetlink::handle_write_event(rofl::cthread &thread, int fd) {
...
-        break;
-      }
-      obj = nl_cache_get_next(obj);
-    }
-
-    thread.add_timer(NL_TIMER_RESYNC, rofl::ctimespec().expire_in(5));
-  } break;
   case NL_TIMER_RESEND_STATE:
+// XXX loop through link cache
+#if 0
```
Comment (DFritzsche) [inline src/roflibs/netlink/cnetlink.cpp]: remove this part?

File: src/roflibs/netlink/cnetlink.cpp [hunk 2]
```diff
@@ -91,44 +95,16 @@ void cnetlink::init_caches() {
         << "cnetlink::init_caches() rtnl_link_alloc_cache_flags failed rc="
         << rc;
   }
-  rc = nl_cache_mngr_add_cache(mngr, caches[NL_NEIGH_CACHE],
-                               (change_func_t)&nl_cb, NULL);
+  rc = nl_cache_mngr_add_cache_v2(mngr, caches[NL_NEIGH_CACHE],
```
Comment (DFritzsche) [inline src/roflibs/netlink/cnetlink.cpp]: is there reason to have a "v2" of this function?

File: src/roflibs/netlink/cnetlink.hpp [hunk 3]
```diff
@@ -161,9 +142,13 @@ class cnetlink : public rofl::cthread_env {
     thread.wakeup();
   }
 
-  void add_neigh_ll(int ifindex, uint16_t vlan, const rofl::caddress_ll &addr);
+#if 0
```
Comment (DFritzsche) [inline src/roflibs/netlink/cnetlink.hpp]: remove?

File: src/roflibs/netlink/ofdpa_bridge.hpp [hunk 4]
```diff
@@ -20,34 +24,33 @@ class ofdpa_bridge {
 
   virtual ~ofdpa_bridge();
 
-  void set_bridge_interface(const rofcore::crtlink &rtl);
+  void set_bridge_interface(rtnl_link *);
 
-  const rofcore::crtlink &get_bridge_interface() const { return bridge; }
+  rtnl_link *get_bridge_interface() const { return bridge; }
```
Comment (DFritzsche) [inline src/roflibs/netlink/ofdpa_bridge.hpp]: why is rv not const?

File: src/roflibs/netlink/cnetlink.cpp [hunk 5]
```diff
@@ -373,144 +335,102 @@ void cnetlink::route_link_apply(int action, const nl_obj &obj) {
...
 
-void cnetlink::route_neigh_apply(int action, const nl_obj &obj) {
-
-  struct rtnl_neigh *neigh = (struct rtnl_neigh *)obj.get_obj();
-
-  int ifindex = rtnl_neigh_get_ifindex(neigh);
-  int family = rtnl_neigh_get_family(neigh);
+void cnetlink::route_neigh_apply(const nl_obj &obj) {
 
+#if 0
```
Comment (DFritzsche) [inline src/roflibs/netlink/cnetlink.cpp]: remove?

File: src/roflibs/netlink/cnetlink.cpp (hunk 1 above)
Comment (toanju) [inline src/roflibs/netlink/cnetlink.cpp]: this should be reimplemented so I would keep it for now

File: src/roflibs/netlink/cnetlink.cpp (hunk 2 above)
Comment (toanju) [inline src/roflibs/netlink/cnetlink.cpp]: the v2 api is the one we want to use starting with this PR

File: src/roflibs/netlink/ofdpa_bridge.hpp (hunk 4 above)
Comment (toanju) [inline src/roflibs/netlink/ofdpa_bridge.hpp]: wasn't used at all so I remove it completely

**Pull request merged**